src/rqt_publisher/publisher.py

```diff
@@ -227,39 +227,59 @@ def _change_publisher_expression(self, publisher_info, topic_name, new_value):
             # Strip topic name from the full topic path
             slot_path = topic_name.replace(publisher_info['topic_name'], '', 1)
             slot_path, slot_array_index = self._extract_array_info(slot_path)
+            slot_type = None
 
-            # Get the property type from the message class
-            slot_type, is_array = \
-                get_slot_type(publisher_info['message_instance'].__class__, slot_path)
-            if slot_array_index is not None:
-                is_array = False
-
-            if is_array:
-                slot_type = list
             # strip possible trailing error message from expression
-            error_prefix = '# error'
-            error_prefix_pos = expression.find(error_prefix)
+            error_prefix = ' # error'
+            error_prefix_pos = user_expression.find(error_prefix)
             if error_prefix_pos >= 0:
-                expression = expression[:error_prefix_pos]
-            success, _ = self._evaluate_expression(expression, slot_type)
+                user_expression = user_expression[:error_prefix_pos]
+
+            computed_expression = str(user_expression)
+            # reduce indexed topic_names, if necessary
+            topic_name_reduction = re.match("(.*)\[[0-9]*\]$", topic_name)
+            if topic_name_reduction:
+                topic_name = topic_name_reduction.group(1)
+
+            # sequences
```

Comment:
Please add a comment describing what this code block is doing and why.

Reply:
Changed to:
```python
# static sequences change one item at a time, impossible to validate
# handle as entire sequence, enables validation
if slot_array_index is not None:
```